source/code/systems/ui_system/private/ui_element.cxx

```diff
@@ -0,0 +1,329 @@
+        out_element.contentbox = out_element.hitbox - padding;
+        out_element.flags = out_element.flags & ~(ElementFlags::Size_AutoWidth | ElementFlags::Size_AutoHeight);
+
+        return any(
+            flags,
+            ElementFlags::Size_StretchWidth
+            | ElementFlags::Size_StretchHeight
+        ) ? UpdateResult::Unresolved : UpdateResult::Resolved;
+    }
+
+    auto element_update_size_stretch(
+        ice::ui::UIData const& data,
+        ice::ui::Element const& parent,
+        ice::ui::ElementInfo const& info,
+        ice::ui::Element& out_element
+    ) noexcept -> ice::ui::UpdateResult
+    {
+        Size size;
+        RectOffset margin;
+        RectOffset padding;
+        ElementFlags flags = ElementFlags::None;
+
+        if (any(parent.flags, ElementFlags::Size_StretchWidth | ElementFlags::Size_StretchHeight))
+        {
+            return UpdateResult::Unresolved;
+        }
+
+        read_size(data, info, size, flags);
+        if (any(flags, ElementFlags::Size_StretchWidth | ElementFlags::Size_StretchHeight) == false)
+        {
+            ICE_ASSERT(false, "Hmmm?!");
+            return UpdateResult::Resolved;
+        }
+
+        read_margin(data, info, margin, flags);
+        read_padding(data, info, padding, flags);
+
+        size = ice::ui::rect_size(out_element.contentbox);
+
+        // We use the hitbox, as we cannot outgrow the 'margin' value of the parent element.
+        ice::ui::Size const parent_size = rect_size(parent.contentbox);
+        if (contains(flags, ElementFlags::Size_StretchWidth))
+        {
+            size.width = parent_size.width - (padding.left + padding.right);
+        }
+        if (contains(flags, ElementFlags::Size_StretchHeight))
+        {
+            size.height = parent_size.height - (padding.top + padding.bottom);
+        }
+
+        out_element.bbox = make_bbox(size, margin, padding);
+        out_element.hitbox = out_element.bbox - margin;// make_hitbox(out_element.bbox, margin);
+        out_element.contentbox = out_element.hitbox - padding;
+        out_element.flags = out_element.flags & ~(ElementFlags::Size_StretchWidth | ElementFlags::Size_StretchHeight);
+        return UpdateResult::Resolved;
+    }
+
+    auto element_update_position(
+        ice::ui::UIData const& data,
+        ice::ui::Element const& parent,
+        ice::ui::ElementInfo const& info,
+        ice::ui::Element& out_element
+    ) noexcept -> ice::ui::UpdateResult
+    {
+        Position position;
+        ElementFlags current_flags = out_element.flags;
+        read_position(data, info, position, current_flags);
+
+
+        // We use the hitbox, as we should start at the 'padded' location of the parent element.
+        Size const parent_size = rect_size(parent.hitbox);
+        Size const size = rect_size(out_element.contentbox); // TOOD: Check if this should not be the hitbox.
```

Comment:
Might be a bug?

Reply:
For now this will be not addressed. As there is no need for the anchoring feature.